Advice App Database Support, Bad Input Handling, and Tests

yusuferdemnacar opened the pull request:
### Requirements in #110, #118, and #120 are met.

- **Database and Statistics**
The app now provides user statistics using the database. The information of the users is kept in the Advice Users table.
GET requests now also returns the statistics to display them on the home page.

- **Erroneous Requests**
The app now handles the erroneous requests. If the request contains bad input (e.g. 187 as age or jg98e as sex etc.) the app responds with status code 400 and the reason for the error as JSON. If the app fails to make a request to the MyHealthFinder API, it responds with status code 502 and the reason for the error as JSON.

- **Unit Tests**
The tests for the GET and POST methods are added. The tests test all possible values for the age, sex, tobaccoUse and sexuallyActive parameters.

**P.S.** The tests take a while (approximately 6 minutes on Intel I7-1165G7) to run because it tries for each age value in the range, so please do not panic if it doesn't immediately finish.

berfinsimsekk reviewed (approved):
I think everything is okay and works correctly. Distribution categories are empty, if there is no user in the database. Maybe you can do something about it but it is okay if you don't want to. Also tests are taking so much time but working correctly. Nice UI, thanks!

haticerk reviewed (approved):
Everything looks fine except the test lasts forever. I think it tests my patience :) . Good job!

yusuferdemnacar commented:
> I think everything is okay and works correctly. Distribution categories are empty, if there is no user in the database. Maybe you can do something about it but it is okay if you don't want to. Also tests are taking so much time but working correctly. Nice UI, thanks!

I added a little message instead of the blank space to the tabs if the user is the first user.



> Everything looks fine except the test lasts forever. I think it tests my patience :) . Good job!

I commented out that test and added 4 new tests to test the boundary values for the age parameter.
